libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -352,16 +352,25 @@ class YulUtilFunctions
 
 	/// @returns the name of a function that allocates memory.
 	/// Modifies the "free memory pointer"
-	/// Arguments: size
-	/// Return value: pointer
+	/// signature: (size) -> ptr
 	std::string allocationFunction();
 
-	/// @returns the name of the function that allocates temporary memory with predefined size
+	/// @returns the name of the function that allocates memory whose size will be defined later,
+	/// or it will be released. The allocation can be finalized using finalizeAllocationFunction.
+	/// Any other allocation will invalidate the memory pointer unless finalizeAllocationFunction
+	/// is called.
 	/// Return value: pointer
-	std::string allocationTemporaryMemoryFunction();
+	/// signature: () -> ptr
+	std::string allocateUnboundedFunction();
+
+	/// @returns the name of the function that finalizes an unbounded memory allocation,
+	/// i.e. sets its size.
+	/// signature: (ptr, size) ->
+	std::string finalizeAllocationFunction();
 
-	/// @returns the name of the function that releases previously allocated temporary memory
-	std::string releaseTemporaryMemoryFunction();
+	/// @returns the name of the function that releases previously allocated memory
+	/// signature: (ptr) ->
+	std::string releaseMemoryFunction();
```

Comment:
Can you release memory allocated by both `allocate()` and `allocate_unbounded()` or only by the former? `allocate_unbounded()` does not even modify the free memory pointer so I'd guess you can't release memory allocated by it but I see in the diff that now `endVisit(FunctionCall)` does just that.

This should be clarified in the docstring because this is pretty arbitrary and depends on how you expect `release_memory()` to work in the future.

Reply:
Maybe it's better to remove the memory release function again.

Reply:
Is the plan to reintroduce it some point though?

Reply:
I also wonder if we might want to preserve the distinction between the normal and temporary allocations. The latter were already paired (correctly - I'm assuming) with deallocations.

Reply:
Maybe, but I think it is rather complicated anyway.